Add flux_led wait method

chocobn69 opened the pull request:
This improve stability, waiting for device to change state beforeanything.

## Description:


**Related issue (if applicable):** fixes #27755 

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.
  - [x] I have followed the [development checklist][dev-checklist]


[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html

homeassistant commented:
Hi @chocobn69, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#29079).

Once you do that we will be able to review and accept this pull request.

Thanks!

homeassistant commented:
Hi @chocobn69, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#29079).

Once you do that we will be able to review and accept this pull request.

Thanks!

balloob commented on homeassistant/components/flux_led/light.py:
```diff
@@ -140,6 +141,11 @@
 
+def flux_led_wait():
+    """Wait device after change state to improve stability."""
+    time.sleep(1)
```
We shouldn't sleep in threads. We should only sleep with `await asyncio.sleep(1)` because it doesn't actually block anything.

chocobn69 replied:
Totally agree, I will turn the whole module async then.

balloob commented on homeassistant/components/flux_led/light.py:
```diff
@@ -298,8 +305,10 @@ def turn_on(self, **kwargs):
                 brightness = self.brightness
             if color_temp > COLOR_TEMP_WARM_VS_COLD_WHITE_CUT_OFF:
                 self._bulb.setRgbw(w=brightness)
+                flux_led_wait()
```
How is this improving things? Isn't the command done when we call `setRgbw` ?

chocobn69 replied:
You can check here : https://github.com/beville/flux_led/blob/master/flux_led/__main__.py
setRgbw does nothing more than sending bytes to a socket: no waiting

chocobn69 commented:
Work to do: 
- [ ] Turn flux_led compoments async
- [ ] Add a parameter `delay` to control waiting time (default 1sec ?)
- [ ] Change documentation according to this

neffs committed:
Limit available heat/cool modes for HomeKit thermostats (#28586)

* Limit available heat/cool modes for HomeKit thermostats.
The Home app only shows appropriate modes (heat/cool/auto) for the device. Depending on the climate integration, disabling the auto start might be needed.

* Include improved mapping for HVAC modes in tests

basnijholt committed:
Sort imports according to PEP8 for opentherm_gw (#29671)

basnijholt committed:
Sort imports according to PEP8 for xiaomi_miio (#29677)

basnijholt committed:
Sort imports according to PEP8 for met (#29699)

basnijholt committed:
Sort imports according to PEP8 for locative (#29698)

basnijholt committed:
Sort imports according to PEP8 for file (#29694)

basnijholt committed:
Sort imports according to PEP8 for ifttt (#29696)

basnijholt committed:
Sort imports according to PEP8 for geofency (#29695)

basnijholt committed:
Sort imports according to PEP8 for counter (#29692)

basnijholt committed:
Sort imports according to PEP8 for ffmpeg (#29693)

basnijholt committed:
use isort to sort imports according to PEP8 for broadlink (#29690)

basnijholt committed:
Sort imports according to PEP8 for ambiclimate (#29689)

basnijholt committed:
Sort imports according to PEP8 for camera (#29691)

basnijholt committed:
use isort to sort imports according to PEP8 for netatmo (#29682)

basnijholt committed:
Sort imports according to PEP8 for specific_devices (#29687)

basnijholt committed:
Sort imports according to PEP8 for wemo (#29685)

basnijholt committed:
Sort imports according to PEP8 for utility_meter (#29710)

basnijholt committed:
Sort imports according to PEP8 for vesync (#29684)

basnijholt committed:
Sort imports according to PEP8 for iaqualink (#29681)

basnijholt committed:
Sort imports according to PEP8 for sensor (#29683)

basnijholt committed:
Sort imports according to PEP8 for ring (#29680)

basnijholt committed:
Sort imports according to PEP8 for http (#29679)

basnijholt committed:
Sort imports according to PEP8 for dyson (#29678)

basnijholt committed:
Sort imports according to PEP8 for somfy (#29675)

basnijholt committed:
Sort imports according to PEP8 for nest (#29670)

basnijholt committed:
Sort imports according to PEP8 for rest (#29674)

basnijholt committed:
Sort imports according to PEP8 for pilight (#29673)

basnijholt committed:
Sort imports according to PEP8 for owntracks (#29672)

basnijholt committed:
Sort imports according to PEP8 for emulated_hue (#29667)

basnijholt committed:
Sort imports according to PEP8 for device_tracker (#29666)

basnijholt committed:
Sort imports according to PEP8 for media_player (#29665)

basnijholt committed:
sort imports according to PEP8 for lock (#29663)

basnijholt committed:
Sort imports according to PEP8 for unifi (#29656)

basnijholt committed:
Sort imports according to PEP8 for deconz (#29659)

basnijholt committed:
Sort imports according to PEP8 for zwave (#29658)

basnijholt committed:
Sort imports according to PEP8 for websocket_api (#29657)

basnijholt committed:
Sort imports according to PEP8 for pushbullet (#29748)

basnijholt committed:
Sort imports according to PEP8 for versasense (#29753)

basnijholt committed:
Sort imports according to PEP8 for remote (#29749)

basnijholt committed:
Sort imports according to PEP8 for yeelight (#29755)

basnijholt committed:
Sort imports according to PEP8 for solarlog (#29752)

basnijholt committed:
Sort imports according to PEP8 for scene (#29750)

basnijholt committed:
Sort imports according to PEP8 for vicare (#29754)

basnijholt committed:
Sort imports according to PEP8 for plaato (#29747)

basnijholt committed:
Sort imports according to PEP8 for shopping_list (#29751)

basnijholt committed:
Sort imports according to PEP8 for emulated_roku (#29756)

basnijholt committed:
Sort imports according to PEP8 for sleepiq (#29759)

basnijholt committed:
Sort imports according to PEP8 for mobile_app (#29758)

basnijholt committed:
Sort imports according to PEP8 for hue (#29757)

basnijholt committed:
Sort imports according to PEP8 for vultr (#29760)

pvizeli committed:
HomeAssistant-pyozw 0.1.7 (#29743)

tetienne committed:
Remove Tahoma component #29744 (#29745)

basnijholt committed:
Sort imports according to PEP8 for components starting with "B" (#29762)

basnijholt committed:
Sort imports according to PEP8 for components starting with "A" (#29761)

basnijholt committed:
Sort imports according to PEP8 for components starting with "E" (#29765)

basnijholt committed:
Sort imports according to PEP8 for components starting with "G" (#29767)

basnijholt committed:
Sort imports according to PEP8 for components starting with "H" (#29768)

basnijholt committed:
use isort to sort imports for components starting with 'm' (#29772)

basnijholt committed:
Sort imports according to PEP8 for components starting with "I" (#29769)

basnijholt committed:
Sort imports according to PEP8 for components starting with "L" (#29771)

basnijholt committed:
Sort imports according to PEP8 for components starting with "K" (#29770)

basnijholt committed:
Sort imports according to PEP8 for components starting with "O" (#29774)

basnijholt committed:
Sort imports according to PEP8 for components starting with "P" (#29775)

basnijholt committed:
Sort imports according to PEP8 for components starting with "S" (#29777)

basnijholt committed:
Sort imports according to PEP8 for components starting with "R" (#29776)

basnijholt committed:
Sort imports according to PEP8 for components starting with "T" (#29778)

basnijholt committed:
Sort imports according to PEP8 for components starting with "V" (#29780)

basnijholt committed:
Sort imports according to PEP8 for components starting with "N" (#29773)

basnijholt committed:
Sort imports according to PEP8 for components starting with "W" (#29781)

basnijholt committed:
Sort imports according to PEP8 for components starting with "Q" (#29785)

basnijholt committed:
Sort imports according to PEP8 for components starting with "Y" (#29783)

basnijholt committed:
Sort imports according to PEP8 for components starting with "X" (#29782)

bramkragten committed:
Updated frontend to 20191204.1 (#29787)

springstan committed:
Move imports to top for homekit (#29560)

* Move imports to top for homekit

* Moved back a couple imports, added annotation to disable import-outside-toplevel

* Fix all tests in test_homekit.py

pvizeli committed:
Cleanup removed component (#29788)

basnijholt committed:
Sort imports according to PEP8 for 'script' folder (#29790)

basnijholt committed:
use isort to sort imports for "setup.py" (#29792)

basnijholt committed:
Sort imports according to PEP8 for 'homeassistant' folder (#29789)

Components are already done

basnijholt committed:
Sort imports according to PEP8 for 'tests' (#29791)

Quentame committed:
Add config flow to iCloud (#28968)

* iCloud: setup ConfigFlow and prepare for more platforms

- add config flow + tests
- fix existing services
- add play_sound & display_message services
- document services
- can use devices with the same name
- prepare to add sensor platform

* Review : not copy account conf

* Review: Safer test patch

* Review: remove reset_account

* Review: Use executor_job while IO

* Review: Use executor_job while IO 2

* Review: use hass.helpers.storage.Store()

* Review: no IO in tests

* Remove reset from services.yaml

* Review: remove authenticate.return_value = Mock()

* Review: do not initialize the api with the mocked service

* isort

* Review: @MartinHjelmare Test config flow with all steps

* Review: Fix failed tests names

* Codevov: Add one missing test

basnijholt committed:
Sort imports according to PEP8 for components starting with "D" (#29764)

* use isort to sort imports for components starting with 'd'

* fix isort mistake

springstan committed:
Move imports to top for ness_alarm (#29518)

* Move imports to top for ness_alarm

* Added patch for the ArminState in alarm_control_panel.py

basnijholt committed:
Sort imports according to PEP8 for components starting with "Z" (#29784)

* use isort to sort imports for components starting with 'z'

* add skip to end of zha/core/channels/__init__.py

* put 'pylint: disable=import-error' at the right place

* remove the import of config_flow in zha/__init__.py

According to @balloob it is no longer needed.

* revert previous commit

* isort:skip homeassistant/components/zha/__init__.py completely

basnijholt committed:
Sort imports according to PEP8 for components starting with "C" (#29763)

basnijholt committed:
Fix 'pytest.register_assert_rewrite("tests.common")' warning (#29797)

see https://github.com/home-assistant/home-assistant/pull/29791/files/fea243c035f117275fda1895720d30e7f977aa9c#r355533273

basnijholt committed:
Sort imports according to PEP8 for huawei_lte (#29664)

basnijholt committed:
Sort imports according to PEP8 for components starting with "U" (#29779)

* use isort to sort imports for components starting with 'u'

* add 'pylint: disable=import-error' to the right place

frenck committed:
Fix build, invalid JSON file in icloud component (#29798)

basnijholt committed:
Sort imports according to PEP8 for google_assistant (#29633)

* sort imports and fix flake8 issue for google

* add isort:skip to EVENT_SYNC_RECEIVED

dmulcahey committed:
Add ZHA group API (#29641)

* add skeleton to retrieve zigbee groups
* get single group
* add a group
* return group members with group
* add comment
* fix group members
* add function to add device to group
* add group members
* add remove from group method
* add api to remove members from group
* add remove groups method
* clean up group add and remove
* fix remove group
* fix remove groups
* add api to get only groupable devices
* change var init
* add tests
* address review comment

Adminiuga committed:
Add input_text reload service. (#29644)

* Add input_text reload service.

* Add test.

Adminiuga committed:
Add input_select reload service. (#29647)

* Add input_select reload service.

* Add test.

frenck committed:
Add JSON files validation to hassfest (#29799)

homeassistant committed:
[ci skip] Translation update

Adminiuga committed:
Fix zha circular import (#29802)

* Refactor zha.core.helpers.
* Make zha isort-able.
* Const import reorg.
* Keep DATA_ZHA config key on entry unload.
* Cleanup ZHA config flow.
* isort.
* Add test.

bwarden committed:
Bump Roku to 4.0.0 (#29809)

ViViDboarder committed:
Nextbus: Sort results for upcoming times (#29811)

Sort upcoming times across all direction's for a particular route and
stop.

frenck committed:
Adjusts repository README (#29805)

balloob committed:
Install discovery requirements if used (#29795)

* Install discovery requirements if used

* Update loader.py

* Fix types

Kane610 committed:
Only update disabled_by when enabled default and entity enabled states differ (#29643)

proferabg committed:
Fix tank utility token (#29801)

* Tank Utility Token Fix

Fix 400 Bad Request on Invalid Token

* Format Fix for Pull Request #29801

* Pylint Annotation Fix for Pull Request #29801

Kane610 committed:
UniFi - Handle disabled switches (#29824)

Kane610 committed:
UniFi - honor detection time when UniFi wire bug happens (#29820)

scop committed:
Re-authorize Huawei LTE on login required error (#29597)

* Re-authorize Huawei LTE on login required error

Closes https://github.com/home-assistant/home-assistant/issues/29531

* Eliminate one indented "else" block

* Convert get_data from inline to instance method

* Use .get instead of [] for checking subscriptions for clarity

vangorra committed:
Fix withings wrong sleep state entry (#29651)

* Fixing issue where wrong sleep state entry was being used. closes #28370,#29397

* Fixing formatting.

* Sorting imports to get black checks to pass.

* Using lambda for getting latest sleep serie.

Adminiuga committed:
Fix input_text initialization with empty config. (#29829)

scop committed:
Upgrade pydocstyle to 5.0.1 (#29830)

* Upgrade pydocstyle to 5.0.1

http://www.pydocstyle.org/en/5.0.1/release_notes.html

* Pydocstyle and other docstring fixes

homeassistant committed:
[ci skip] Translation update

basnijholt committed:
Sort imports for requirements.py and its test using isort (#29836)

Unblocks https://github.com/home-assistant/home-assistant/pull/29739

ratsept committed:
Fix Soma integration connection issue (#27692)

* Added a check for Connect actually returning something before telling the user the setup succeeded

* Added handling for KeyError in case API returns empty response, formatting

* Trying to please the linter

pvizeli committed:
Remove uvloop event policy (#29835)

* Remove uvloop event policy

* Clean tests

* Fix lint

* Cleanup statment

springstan committed:
Add user-agent to fix dwd_weather_warnings setup error (#29596)

* Added dummy user-agent to http request to fix setup error

* Replace dummy user-agent with the user-agent of the global home assistant session

* Adjust comment and rename user-agent constant

springstan committed:
Add user-agent to fix dwd_weather_warnings setup error (#29596)

* Added dummy user-agent to http request to fix setup error

* Replace dummy user-agent with the user-agent of the global home assistant session

* Adjust comment and rename user-agent constant

Kane610 committed:
Add more logging to help future debug situations (#29800)

foxy82 committed:
Update rfxtrx component so it can be run as a custom_component (#29638)

* Updating rfxtrx component so it can be run as a custom_component

* Fix pylint errors

* Fix pylint errors

* Fix pylint error on dict.

* isort

homeassistant committed:
[ci skip] Translation update

balloob committed:
Revert Tahoma removal (#29840)

* Revert "Remove Tahoma component #29744 (#29745)"

This reverts commit df74272ba6311d527fd07198929c80a45d9fed15.

* Revert "Cleanup removed component (#29788)"

This reverts commit 3a28361bebbe11d124793a0ec269de74ef180665.

scop committed:
Use Bionic's ffmpeg on Travis, jonathonf/ffmpeg-4 is N/A at the moment (#29860)

bachya committed:
Bump aioambient to 1.0.2 (#29850)

frenck committed:
Fix package import sort on dwd_weather_warnings (#29874)

basnijholt committed:
Add isort to CI and pre-commit (#29739)

* add isort to CI and pre-commit

* disable wrong-import-order in pylintrc

* :pencil2: Tweak


Co-authored-by: Franck Nijhof <frenck@frenck.nl>

scop committed:
Remove no longer needed auth.util, use secrets instead (#29861)

Adminiuga committed:
Log ZHA bind/unbind operations status (#29842)

* Log bind/unbind operation result.
* Use ZDO consts.
* Use device logger for bind/unbind results.
* Lint.

johnluetke committed:
(Re)Add support for multiple Pi-Holes (#27569)

* Update configuration schema to support multiple Pi-holes

* Construct sensors for each configured Pi-hole

* Ensure each Pi-hole has a unique name

* Update services to handle multiple Pi-holes

* Update tests for multiple configurations

* Refactor tests to support service testing

* Fix else-raise per pyliunt

* Per code review, add all entities in a single call

* Per code review, add the default name as default.

* Per code review, add cv.ensure_list to prevent breaking change

* Per code review, move name validation to schema

* Remove default name

* Per code review, validate api_key in schema definition

* Per code review, rename variables

* Per code review, use list comprehension

* Ensure unique slug names in config validation

* Per code review, refactor to CoroutineMock

* Fix adding sensor entities

* Per code review, refactor mock function creation

* Per code review, refactor mock function return values

randellhodges committed:
ISY994 Node Filter Update (#28155)

* Adding filters for keypad dimmers and smoke sensor

* ran black --fast as suggested

* Adding filters for keypad dimmers and smoke sensor

* ran black --fast as suggested

Adminiuga committed:
Defer log formatting. (#29888)

syssi committed:
Bump python-miio version to 0.4.8 (#29890)

homeassistant committed:
[ci skip] Translation update

springstan committed:
Fix setup for tank_utility component (#29902)

scop committed:
Make Python deprecation notice easier to maintain (#29900)

fabaff committed:
Upgrade Sphinx to 2.2.2 and sphinx-autodoc-typehintsi to 1.10.3 (#29906)

frenck committed:
Fix incorrect file format yr test fixure (#29910)

frenck committed:
Fixes invalid JSON syntax in devcontainer (#29911)

springstan committed:
Fix setup error for logbook (#29908)

* Fix setup error by moving an import back into the setup function

* Revert c741664d4da76611b5bb7502dda61a24dce22c61

* Add homekit as after_dependency to logbook manifest.json

balloob committed:
Add integration platform helper (#29914)

homeassistant committed:
[ci skip] Translation update

JBassett committed:
Fix mobile app device identifiers (#29920)

Fix identifiers when updating device registration.

amelchio committed:
Support entity_id: all in lifx.set_state (#29919)

DjMoren committed:
Update Tahoma component's tahoma-api requirement's version (#29918)

ochlocracy committed:
Implement Alexa.EventDetectionSensor for Alexa  (#28276)

* Implement Alexa.EventDetectionSensor Interface

* Removed references to PR #28218 not yet merged into dev.

* Update tests to include Alexa Interface

* Guard for `unknown` and `unavailible` states.

* Fixed Unnecessary "elif" after "return"

raman325 committed:
Update androidtv version to improve source selection support (#29579)

* Change androidtv module versions and add support for select_source for all device types

* Update and add tests

* Update requirements_test_all.txt

* Update requirements_all.txt

* Consolidate tests

* Fix typo

* Remove 'self._device'

Quentame committed:
Add battery sensor to iCloud (#29818)

* Add battery sensor to iCloud

* Update .coveragerc

* Review: @balloob & @MartinHjelmare

* Review: use f string

ChrisMandich committed:
Fix loading flume integration (#29926)

* Fix https://github.com/home-assistant/home-assistant/issues/29853

* Run script.gen_requirements

* Update to store Token File in config directory

* Update to store Token File in config directory

* Update to store Token File in config directory

flamechair committed:
bump venstar 0.12 (#29954)

* bump venstar 0.12

* Update manifest.json

JeffLIrion committed:
Bump adb-shell to 0.1.0 and androidtv to 0.0.36 (#29938)

* Bump adb-shell to 0.1.0 and androidtv to 0.0.36

* Add test for setting up two devices

* Add test_setup_same_device_twice

* Fix test_setup_two_devices

* Fix coverage

* Coverage

* Fix flaky 'test_setup_two_devices'

* Another stab at coverage

* Rename 'address' back to 'host'

elupus committed:
Start of using hass state for tests rather than direct object (#29377)

caronc committed:
isort fix on test_media_player (#29965)

fabaff committed:
Upgrade keyring to 20.0.0 and keyrings.alt to 3.4.0 (#29960)

boralyl committed:
Fix example value for Todoist service (#29953)

scop committed:
Remove deprecated rflink configs (#29972)

They've been deprecated and automatically replaced since July 2017
already, fe6a4b8ae5953c8d565dc93acc9d34c715a80aee

homeassistant committed:
[ci skip] Translation update

oandrew committed:
Fix support for legacy Z-Wave thermostats (#29955)

This brings back support for Z-Wave thermostats
of SETPOINT_THERMOSTAT specific device class.
Such devices don't have COMMAND_CLASS_THERMOSTAT_MODE
and are now handled separately.

scop committed:
Drop Python 3.6 support (#29978)

elupus committed:
Support case of unknown/unavailable temperature/humidity (#29959)

* Support case of unknown/unavailable temperature/humidity

State is never None, just a string.

* Lint suggestion

eifinger committed:
Migrate to api_key (#29966)

JeffLIrion committed:
Remove 'SUPPORT_PLAY_MEDIA' from Volumio (#29969)

balloob committed:
Make hassfest import detection better (#29932)

* Make hassfest import detection better

* Fix tests

rsnodgrass committed:
Fixed "condtion_type" to "condition_type" (#29984)

Fixed "condtion_type" to "condition_type"

balloob committed:
Fix condition typo (#29989)

lddubeau committed:
Don't use the locals parameter on exec. (#29979)

Using the locals parameter makes it so that the code of a Python script runs as
if it were in the body of a ``class``. One effect of this is that functions
defined as part of a script cannot call one another directly.

Fixes: #24704, #13653

Emacee committed:
Update binary_sensor.py (#29977)

Change sensor type for central locking from safety into lock.

frenck committed:
Bump shodan to 1.21.0 (#29991)

frenck committed:
Bump pytest to 5.3.2 (#29990)

frenck committed:
Add check-json to CI and Pre-commit (#29912)

* Add check-json to CI and Pre-commit

* Add ignore pre-commit hooks to gen_requirements_all

balloob committed:
Add unique ID to config entries (#29806)

* Add unique ID to config entries

* Unload existing entries with same unique ID if flow with unique ID is
finished

* Remove unused exception

* Fix typing

* silence pylint

* Fix tests

* Add unique ID to Hue

* Address typing comment

* Tweaks to comments

* lint

zewelor committed:
Show current effect in yeelight device (#28975)

* Show current effect in yeelight device

* Use device_state_attributes instead of state_attributes

* Add early return in set effect

* Make single if elif chain

* Fix if elif

* Fix if elif

balloob committed:
Convert Hue to use unique ID (#30000)

* Convert Hue to use unique ID

* Fix normalization

* Store/restore unique ID

* Fix tests

vzahradnik committed:
Fix modbus service description (#30005)

frenck committed:
Fix persistent setup error notification content (#29995)

* Fix persistent setup error notification content

* Use documentation from manifest, enriched error messages

* Fix issue caught by mypy

homeassistant committed:
[ci skip] Translation update

vangorra committed:
Make tplink light more responsive (#28652)

* Making tplink light more responsive.

* Adding light platform tests.

* Addressing PR feedback.

* Mocking the module, not the api.

* Using sync method for background update.

andre-richter committed:
Z-Wave: Fibaro FGR*-222: Add venetian blind support (#29701)

The Fibaro FGR-222/FGRM-2222 ZWave roller shutter devices have a proprietary
command class to support setting the tilt angle of venetian blinds (= type of
window cover). This PR adds the support to HA for this. This allows the user to
set the height of the blinds and the tilt angle separately.

Original patch by @ChristianKuehnel #24405.

fabaff committed:
Upgrade matrix-client to 0.3.2 (#30027)

fabaff committed:
Upgrade zeroconf to 0.24.1 (#30028)

homeassistant committed:
[ci skip] Translation update

gtdiehl committed:
Update Envoy sensor to configure credentials and grab Inverter Date from updated API (#28837)

* Update manifest.json

Updated sensor to use the latest version of the API to be able to use the new features.

* Updated sensor to use new API features.  Configurable credentials and Inverter last reported date.

* EnvoyReader is passed to Envoy uses async update()

* Fixed pydocstyle warnings

* Fixed merge issue. Had extra variable.

* Added warning message when authentication for Inverter data fails

* Added continue after exception for loop

* Moved if statement outside of try/except

* Removed unneeded boolean attribute

balloob committed:
Add option to ignore flows (#30008)

erikkastelec committed:
changed Venstar component temperature to half degree accuracy (#30034)

omriasta committed:
Patch rachio (#30031)

* fix binary sensor offline/online

fixed self._handle_update on line 66 to produce args, kwargs.
Updated the binary sensor to check the correct index in the tuple.

* Fixed Standby switch

Set standby switch to poll in order to get status when homeassistant starts up.
Updated the index for the switch to get the status from the tuple.

bachya committed:
Bump simplisafe-python to 5.3.6 (#30055)

Adminiuga committed:
Add timer reload service. (#30015)

Commit:
Fix failure in transform method (#30023)

* Fix failure in transform method

* Fix formatting issue

homeassistant committed:
[ci skip] Translation update

bachya committed:
Guard against future unknown SimpliSafe entity types (#30059)

* Guard against future unknown SimpliSafe entity types

* Updated log message

Jc2k committed:
Set unique id on homekit_controller config entries (#30035)

* Set unique id on config entries

* Changes from review

balloob committed:
Remove stream from camera after deps (#30057)

yegle committed:
Add unit_of_measurement to various Transmission sensors (#30037)

* Add unit_of_measurement to various Transmission sensors 

Without unit_of_measurement, the history graph card will not show those sensors as line chart.

* Change Counts to Torrents.

ochlocracy committed:
Refactor Alexa capabilityResources object into class, Implement Alexa semantics object (#29917)

* Refactor capabilityResources object into class.
Implement semantics object to support open, close, raise, lower utterences.
Replace covers PercentageController with RangeController.
Add semantics for covers.
Remove PowerController for covers.
Add new display categories.
Add new items to Alexa Global Catalog.
Implement garage door voice PIN code support though Alexa app.
Fixed bug with getting property for ModeController.
Fixed bug were PercentageController AdjustPercentage would exceed 100.

* Comment fixes in Tests.

* Reorder imports.

* Added additional tests for more code coverage.

* Added and additional test for more code coverage.

* Explicitly return None for configuration() if not instance of AlexaCapabilityResource.

balloob committed:
Make Hassfest stricter pt 2 (#30068)

* Make Hassfest stricter

* Fix if-condition

* Small cleanup

bieniu committed:
Init entities as unavailable when offline (#29738)

springstan committed:
Make name of nmbs live sensor customizable via unique_id (#29662)

* Make name customizable for nmbs live sensors

* Change order of the live name

* Added unique id to live nmbs sensor

* Revert changing the name of the live sensor

scop committed:
Include all SSDP data in discovery info (#28197)

* Include all SSDP data in discovery info

* Use UPnP device description as discovery info, inject some SSDP attrs

* Clean up attribute names

* Adapt existing SSDP flows to changed attribute names

* Prefix all SSDP UPnP attribute name constants with ATTR_UPNP, tweak a bit

homeassistant committed:
[ci skip] Translation update

frenck committed:
Fix update port and api key on deconz discovery config entry u… (#30088)

* Fix update port and api key on discovery config entry update

* Remove coroutine from _update_entry

fuzzie360 committed:
Fix homekit handling of 0 light brightness and fan speed (#29962)

* Fix homekit handling of 0 light brightness and fan speed

* Update homekit tests for new initial brightness/speed value

michaelarnauts committed:
Rework tado component (#29246)

* Fix imports so it works in custom_components

* Rework tado component

* Code cleanup

* Remove water_heater

* Address pylint warnings

* Remove water_heater from components

* Raise PlatformNotReady when we couldn't connect

* Revert PlatformNotReady since we are not a platform

* Add debugging information

* Add fallback setting

* Import with relative path

* Address race condition

* Cleanup

* Catch 422 Errors and log the real error

* Use async_schedule_update_ha_state to update the entities

* Forgot the True

gppanayotov committed:
Add an open window sensor for heating zones (#30090)

ochlocracy committed:
Explicitly include Alexa interface for image_processing entities. (#30101)

springstan committed:
Bump starlingbank to 3.2 (#30098)

balloob committed:
Bump ring to 0.2.5 (#30103)

Jc2k committed:
Don't error on removal of an ignored homekit_controller config entry (#30083)

* Don't error on ignored entries

* Don't ever call async_remove_entry or async_unload_entry for an unignored ignore config entry

basnijholt committed:
add --show-diff-on-failure to pre-commit (#30097)

This makes the traceback on a failing CI pipeline much more useful.

scop committed:
Upgrade mypy to 0.761 (#30104)

https://mypy-lang.blogspot.com/2019/12/mypy-0760-released.html
https://github.com/python/mypy/releases/tag/v0.761

scop committed:
Helpers type hint improvements (#30106)

vzahradnik committed:
Add device class attribute to modbus sensors (#30030)

cereal2nd committed:
Upgrade python-velbus (#30110)

Jc2k committed:
Add a config entry mechanism to rediscover a discovery that was ignored (#30099)

* Mechanism to rediscover a discovery that was ignored

* Add core config entry tests for new rediscover step

* Add tests for homekit_controller implementation of async_step_rediscover

* Rename rediscover to unignore

* Comment the new ignore/unignore mechanisms

Adminiuga committed:
Implement ZHA entity classes registry (#30108)

* ZHA Entity registry.
Match a zha_device and channels to a ZHA entity.

* Refactor ZHA sensor to use registry.
* Remove sensor_types registry.
* Fix ZHA device tracker battery remaining.
* Remove should_poll/force_update attributes.
* Fix binary_sensor regression.
* isort.
* Pylint.
* Don't access protected members.
* Address comments and fix spelling.
* Make pylint happy again.

balloob committed:
Clean up mobile app webhooks (#30123)

Adminiuga committed:
Refactor ZHA electrical measurement sensor. (#30130)

Adminiuga committed:
Cleanup ZHAEntity class (#30131)

Remove `_domain` attribute since we're not using it anymore.

springstan committed:
Move imports into setup function in homekit __init__.py (#30137)

balloob committed:
Fix test

balloob committed:
Remove requirement from entity integration (#30113)

fabaff committed:
Upgrade zeroconf to 0.24.2 (#30140)

Adminiuga committed:
Refactor ZHA binary_sensor (#30138)

* Refactor ZHA binary_sensor.

Use ZHA entity class registry for channel specific implementations.

* Remove registries.BINARY_SENSOR_TYPES dict.

* Address PR comments.

dannerph committed:
Update Integration of Keba charging station (#30125)

* fixed parsing of current to float in service set_current

* Added optional name in the config file in order to get a better entety naming (easier to find)

* fix parsing of all parameters to service calls

* addressed code review comments + updated pypi dependency

* config name imported from cont.py + minor naming changes to be more clear about the meaning of a sensor

* removed name in config again, use product name gathered from the charging station instead

* implemented suggested changes

* changed variable naming as requested

scop committed:
More helpers type improvements (#30145)

frenck committed:
Bump sqlalchemy to 1.3.12 (#30142)

cgtobi committed:
Clean up scaffold (#30135)

Adminiuga committed:
ZHA binary_sensor cleanup. (#30149)

frenck committed:
Fix deconz SSDP updating Hassio discovery (#30153)

Kane610 committed:
Allow battery value of 0 as well as make sure to not create a battery tracker if one already exist (#30155)

ochlocracy committed:
Add support for input_number entities in Alexa integration (#30139)

* Add support for input_number entities

* Update homeassistant/components/alexa/capabilities.py

Co-Authored-By: Paulus Schoutsen <paulus@home-assistant.io>

* Removed get methods to directly access required attributes dicts.

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

Quentame committed:
Add icon to Plex sensor (#30172)

springstan committed:
Add cast to state of Dyson Air Quality Sensor (#30100)

Quentame committed:
Add icons to Freebox sensors (#30132)

gtdiehl committed:
Upgrade envoy_reader to 0.11.0 (#30179)

P-Verbrugge committed:
Updated formatting of total_blocks value (#30170)

The number of total blocks is always a round number. There can't be .1 or .11 blocks for example. The output is now always formatted with two decimals that are always 00.

heckler committed:
New date_time_utc display option added to the time_date sensor platform (#30158)

alandtse committed:
Enable config flow for Tesla (#28744)

* build: bump teslajsonpy to 0.2.0

* Remove tests

* feat: add config flow

* feat: add async

* perf: convert unnecessary async calls to sync

* feat: add charger voltage and current sensor

* feat: add options flow

* build: bump teslajsonpy to 0.2.0

* Remove icon property

* Revert climate mode change

* Remove charger sensor

* Simplify async_setup_platform

* Update homeassistant/components/tesla/sensor.py

Co-Authored-By: Paulus Schoutsen <paulus@home-assistant.io>

* Update homeassistant/components/tesla/binary_sensor.py

Co-Authored-By: Paulus Schoutsen <paulus@home-assistant.io>

* Address requested changes

* Fix pylint error

* Address requested changes

* Update codeowners

* Fix pylint error

* Address requested changes

* Address requested change

* Remove unnecessary check for existing config entry

* Load scan_interval in async_setup_entry

* Include coverage of config_flow

* Add tests for full coverage

* Address requested test changes

* Remove unnecessary init lines

* Remove unnecessary init

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

dmulcahey committed:
Protect against bad data stored in ZHA (#30183)

homeassistant committed:
[ci skip] Translation update

fabaff committed:
Upgrade zeroconf to 0.24.3 (#30187)

Quentame committed:
Fix AdGuard Home safe search sensor name (#30171)

* Fix AdGuard Home safe search sensor name

* Fix typo

garbled1 committed:
Add onewire devices and owserver remote host support (#29948)

* Add support for HobbyBoards sensors: Hub, Humidity, Moisture, PulseCounter
Add support for an owserver running on a remote host

* Cleanup and fixes for style/etc

* Forgot to convert to f-strings.

* Update with changes requested by @MartinHjelmare

* Pylint

* Fix to None

* Reverse logic for hobbyboard test to eliminate pylint complaint

* Modified manifest to list myself as codeowner

* Move some functions down into OneWireProxy instead of the top OneWire

* Oops, missed some decode()'s

* And another!

springstan committed:
Bump importlib-metadata to 1.3.0 (#30196)

* Bump importlib-metadata to 1.3.0

* Include setup.py and package_constraints.txt

springstan committed:
Bump datapoint to 0.9.5 (#30185)

ochlocracy committed:
Implement EqualizerController in Alexa for media_player. (#30159)

homeassistant committed:
[ci skip] Translation update

BKPepe committed:
Upgrade youtube_dl to 2019.12.25 (#30203)

ndonegan committed:
Pull track position from MPD status (#28579)

* Pull track position from MPD status()

This allows the progress bar to work when using the media-control card in lovelace.

* Actually commit flake8 fix!

* Extra documentation.

Mainly to trigger CI rerun.

* Updated to use self._media_position

rhadamantys committed:
Add support for enocean window handle FA 10 00 (Hoppe) (#29968)

* Added support for Somfy RTS wireless power socket and
Somfy Temperature Sensore Thermos Wirefree io

* Added code formatting fixes for commit 5faaf9c

* added support for RollerShutterRTSComponent from Somfy

* Added support for RTS roller shutter in set_cover_position

* added support for enocean window handle FA 10 00 (Hoppe)

* removed spaces in empty lines

* removal of rawdata attribute / code style

* isort fix

* PyLint fixes

* Improvements after review suggestions: rename device class to sensor type to avoid ambiguousness

* added import for DEVICE_CLASS_POWER from const.py

* removed window handle unit

HarlemSquirrel committed:
Update Vivotek camera component (#30191)

- Bump libpyvivotek version to 0.4.0
- Add digest authentication

scop committed:
Make Huawei LTE notify service name configurable (#30208)

* Default Huawei LTE notify service name to notify.huawei_lte, make configurable

Closes https://github.com/home-assistant/home-assistant/issues/29409

* Set default notify service name for uninvoked options flow

scop committed:
Add Huawei LTE integration suspend and resume services (#30207)

Useful e.g. if accessing the router web interface from another source
such as a web browser is temporarily required.

homeassistant committed:
[ci skip] Translation update

alandtse committed:
Bump teslajsonpy to 0.2.1 (#30217)

Closes #29922

PaulAnnekov committed:
bump tuyaha 0.0.5 (#30213)

bendavid committed:
Fix handling of symlinked device descriptors in keyboard_remote and move remaining sync io to executor thread pool (#30206)

* fix handling of symlinked device decriptors

* make check for symlinked paths more efficient

* make variable names pylint compliant

* move sync io during setup and device connect/disconnect to executor thread pool

* move remaining sync io during setup to executor thread pool

* remove unnecessary lambda functions
